File: equo-app-angular/src/main/java/samplename/MyEventHandler.java
package samplename;

import org.osgi.service.component.annotations.Component;
import org.osgi.service.component.annotations.Reference;

import com.equo.comm.api.IEquoEventHandler;
import com.equo.comm.api.actions.IActionHandler;
import com.equo.comm.api.annotations.EventName;

@Component
public class MyEventHandler implements IActionHandler {

	@Reference
	private IEquoEventHandler equoEventHandler;

	@EventName("MyEventHandler")
	public String myFirstEvent(String payload) {
		System.out.println("First event: " + payload);
		equoEventHandler.send("MyJavaEvent", "This is your first message received from Java");
		return payload;
	}
}

File: kotlin-react/src/main/kotlin/samplename/MyEventHandler.kt
package samplename

import com.equo.comm.api.IEquoEventHandler
import com.equo.comm.api.actions.IActionHandler
import com.equo.comm.api.annotations.EventName
import org.osgi.service.component.annotations.Component
import org.osgi.service.component.annotations.Reference

@Component
class MyEventHandler : IActionHandler {

    @Reference
    private lateinit var equoEventHandler: IEquoEventHandler

    @EventName("MyEventHandler")
    fun myFirstEvent(payload: String): String {
        println("First event: $payload");
        equoEventHandler.send("MyKotlinEvent", "This is your first message received from Kotlin");
        return payload
    }
}